src/main/java/graphql/annotations/processor/GraphQLAnnotations.java

```diff
@@ -0,0 +1,145 @@
+import graphql.annotations.processor.graphQLProcessors.GraphQLAnnotationsProcessor;
+import graphql.annotations.processor.graphQLProcessors.GraphQLInputProcessor;
+import graphql.annotations.processor.graphQLProcessors.GraphQLOutputProcessor;
+import graphql.annotations.processor.typeFunctions.DefaultTypeFunction;
+import graphql.annotations.processor.typeFunctions.TypeFunction;
+import graphql.annotations.processor.retrievers.GraphQLObjectInfoRetriever;
+import graphql.relay.Relay;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLObjectType;
+import org.osgi.service.component.annotations.Component;
+
+import java.util.*;
+
+import static graphql.annotations.processor.util.NamingKit.toGraphqlName;
+
+/**
+ * A utility class for extracting GraphQL data structures from annotated
+ * elements.
+ */
+@Component
+public class GraphQLAnnotations implements GraphQLAnnotationsProcessor {
+
+    private GraphQLObjectHandler graphQLObjectHandler;
+    private ProcessingElementsContainer container;
+
+    private GraphQLObjectInfoRetriever graphQLObjectInfoRetriever;
+
+    public GraphQLAnnotations() {
+        this(new DefaultTypeFunction(new GraphQLInputProcessor(), new GraphQLOutputProcessor()), new GraphQLObjectInfoRetriever(), new GraphQLObjectHandler());
+    }
+
+    public GraphQLAnnotations(TypeFunction defaultTypeFunction, GraphQLObjectInfoRetriever graphQLObjectInfoRetriever, GraphQLObjectHandler graphQLObjectHandler) {
+        this.graphQLObjectInfoRetriever = graphQLObjectInfoRetriever;
+        this.defaultTypeFunction = defaultTypeFunction;
+        this.graphQLObjectHandler = graphQLObjectHandler;
+        this.container = initializeContainer(this.defaultTypeFunction);
+    }
+
+    private ProcessingElementsContainer initializeContainer(TypeFunction defaultTypeFunction) {
+        Map<String, graphql.schema.GraphQLType> typeRegistry = new HashMap<>();
+        Map<Class<?>, Set<Class<?>>> extensionsTypeRegistry = new HashMap<>();
+        final Stack<String> processing = new Stack<>();
+        Relay relay = new Relay();
+        ProcessingElementsContainer container = new ProcessingElementsContainer(defaultTypeFunction, relay, typeRegistry, extensionsTypeRegistry, processing);
+        return container;
+    }
+
+    public static GraphQLAnnotations instance = new GraphQLAnnotations();
+
+    public static GraphQLAnnotations getInstance() {
+        return instance;
+    }
+
+    public void setRelay(Relay relay) {
+        this.container.setRelay(relay);
+    }
+
+
+    public String getTypeName(Class<?> objectClass) {
+        GraphQLName name = objectClass.getAnnotation(GraphQLName.class);
+        return toGraphqlName(name == null ? objectClass.getSimpleName() : name.value());
+    }
+
+    public static GraphQLObjectType object(Class<?> object) throws GraphQLAnnotationsException {
+        return new GraphQLObjectHandler().getObject(object, getInstance().getContainer());
```

Comment:
Could we avoid creating new instance of GraphQLObjectHandler everytime we create an object ?